include/openssl/crypto.h

```diff
@@ -245,6 +245,55 @@ typedef struct {
     struct CRYPTO_dynlock_value *data;
 } CRYPTO_dynlock;
 
+# if !defined(OPENSSL_THREADS) || defined(CRYPTO_TDEBUG)
+typedef unsigned int CRYPTO_RWLOCK;
+typedef unsigned int CRYPTO_ONCE;
+typedef unsigned int CRYPTO_THREAD_LOCAL;
+typedef unsigned int CRYPTO_THREAD_ID;
+
+#  define CRYPTO_ONCE_STATIC_INIT 0
+# elif defined(OPENSSL_SYS_WINDOWS)
+#  include <windows.h>
+typedef CRITICAL_SECTION CRYPTO_RWLOCK;
+typedef DWORD CRYPTO_THREAD_LOCAL;
+typedef DWORD CRYPTO_THREAD_ID;
```

Comment:
Do you want to consider:

- typedef unsigned int/HANDLE/pthread_t CRYPTO_THREAD
- CRYPTO_THREAD CRYPTO_THREAD_get_current()
- int CRYPTO_THREAD_compare (CRYPTO_THREAD a, CRYPTO_THREAD b) -> in windows it gets thread id and compare.

and if required:

- typedef unsigned int/DWORD/pthread_t CRYPTO_THREAD_ID
- only in windows CRYPTO_THREAD and CRYPTO_THREAD_ID are different in other two cases both are same.  

So that CRYPTO_THREAD type can be used in the applications without explicitly defined.

my 2 cents.